### PR TITLE
Allow to request resending of password reset code

### PR DESCRIPTION
**Pull request by Devessier**

### CODE REVIEW

File: packages/client/__tests__/AuthenticationPasswordReset.test.tsx [hunk 1]
```diff
@@ -967,6 +1046,134 @@ cases<{
...
+        'Rendering password reset code screen': {
+            'Requested password reset code resending':
+                | 'Displaying reached rate limit toast'
+                | 'Displaying success toast'
+                | 'Displaying unknown error toast';
+        };
+    };
+    events: EventFrom<typeof passwordResetMachine>[];
+}>(
+    'Request password reset email resending',
```
Comment (prastoin) [inline packages/client/__tests__/AuthenticationPasswordReset.test.tsx:1060]: **🤓 nitpick:** Should we update test group label ?

File: packages/client/__tests__/AuthenticationPasswordReset.test.tsx [hunk 2]
```diff
@@ -335,8 +338,55 @@ const passwordResetMachine =
                                 },
                             },
                         },
-                        'Token validated': {
-                            type: 'final',
+                        'Requested password reset code resending': {
```
Comment (prastoin) [inline packages/client/__tests__/AuthenticationPasswordReset.test.tsx:341]: **👏 praise:** Really nice !

**Review by prastoin — APPROVED** [review]
Great work !
It's now really common for us to be testing anything in the client using xstate test !
Raised a nitpick
Still able to be merged

File: packages/client/__tests__/AuthenticationPasswordReset.test.tsx (hunk 1 above)
Comment (Devessier) [reply packages/client/__tests__/AuthenticationPasswordReset.test.tsx:1060]: How should we rename the group? These tests are about resending an email for password reset.

Comment (prastoin) [reply packages/client/__tests__/AuthenticationPasswordReset.test.tsx:1060]: Indeed ! My bad nevermind

**Review by prastoin — APPROVED** [review]
Post master rebase re review
Ready to be merged !

**Pull request merged**